### PR TITLE
add unit test to test date formatting

### PR DESCRIPTION
**Pull request by rianfowler**
## Description
- add unit tests for broken date formatting 

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 1]
```diff
@@ -37,9 +38,11 @@ class SaveInProgressIntro extends React.Component {
     );
     if (login.currentlyLoggedIn) {
       if (savedForm) {
+        const lastUpdated =
+          savedForm.lastUpdated || _.get('metadata.lastUpdated', savedForm);
         const savedAt = this.props.lastSavedDate
           ? moment(this.props.lastSavedDate)
-          : moment.unix(savedForm.lastUpdated);
```
Comment (rianfowler) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: the component is using `savedForm.lastUpdated` but the unit test looks for `savedForm.metadata.lastUpdated`. I don't know which one of these is correct so I added some logic to use the metadata if the `savedForm.lastUpdated` doesn't exist.

Comment (jbalboni) [reply src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: I think that's ok. The different dates in here are confusing, I'm not sure why we have so many similar ones.

**Review by jbalboni — APPROVED** [review]

File: src/platform/forms/save-in-progress/SaveInProgressIntro.jsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import moment from 'moment';
 
 import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 import { getNextPagePath } from 'us-forms-system/lib/js/routing';
+import _ from '../../../platform/utilities/data';
```
Comment (erikphansen) [inline src/platform/forms/save-in-progress/SaveInProgressIntro.jsx]: you don't need the `../../../` here

File: src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx [hunk 3]
```diff
@@ -50,6 +53,10 @@ describe('Schemaform <SaveInProgressIntro>', () => {
       />,
     );
 
+    expect(
+      tree.find('.saved-form-item-metadata').get(1).props.children[1],
+    ).to.equal(moment.unix(946684800).format('M/D/YYYY [at] h:mm a'));
```
Comment (erikphansen) [inline src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx]: I'd love to hear your thoughts on this, but my instinct would be to hard code the formatted date we are expecting here. In this case it's _highly_ unlikely that Moment will get wonky in a future version, but if it did this test wouldn't catch it.

**Review by erikphansen — APPROVED** [review]

File: src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx (hunk 3 above)
Comment (rianfowler) [reply src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx]: I don't know what timezone to code it for. what I code for in Utah will probably fail on servers in Virginia.

Comment (erikphansen) [reply src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx]: heh, totally valid point!

Comment (rianfowler) [reply src/platform/forms/tests/save-in-progress/SaveInProgressIntro.unit.spec.jsx]: I think this implementation could be improved if we didn't have components creating data. would be easier to test if it was just a pure function whose value was injected through a prop.

**Pull request merged**